radicle-node/src/service/session.rs

```diff
@@ -18,10 +20,10 @@ pub enum PingState {
 }
 
 /// Session protocol.
-#[derive(Debug, Copy, PartialEq, Eq, Clone)]
+#[derive(Debug, PartialEq, Eq, Clone)]
 pub enum Protocol {
     /// The default message-based gossip protocol.
-    Gossip { requested: Option<Id> },
+    Gossip { requested: Option<(Id, Namespaces)> },
```

Comment:
I think I need some explanation on why `Namespaces` is being shoved in here :thinking: